esmvaltool/diag_scripts/iht_toa/single_model_diagnostics.py

```diff
@@ -0,0 +1,714 @@
+"""(C) Crown Copyright 2023, the Met Office.
+
+Single model diagnostics
+1. Solve the Poisson solver
+2. Produce and save plots
+"""
+
+import datetime
+import logging
+from copy import deepcopy
+from pathlib import Path
+
+import cartopy.crs as ccrs
+import iris
+import iris.plot as iplt
+import matplotlib.dates as mdates
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib import gridspec, rcParams
+from poisson_solver import SphericalPoisson
+
+from esmvaltool.diag_scripts.shared import (
+    group_metadata,
+    run_diagnostic,
+    save_figure,
+)
+
+# Initialise logger
+logger = logging.getLogger(Path(__file__).stem)
+
+rcParams.update({
+    'font.size': 14,
+    'xtick.major.pad': 10,
+    'ytick.major.pad': 10,
+    'xtick.major.size': 10,
+    'ytick.major.size': 10,
+    'xtick.minor.size': 5,
+    'ytick.minor.size': 5,
+    'axes.linewidth': 2,
+    'lines.markersize': 8,
+    'lines.linewidth': 2
+})
+
+
+def get_provenance_record(plot_type, ancestor_files):
+    """Create a provenance record describing the diagnostic data and plot."""
+    record = {
+        'statistics': ['mean'],
+        'domains': ['global'],
+        'plot_types': [plot_type],
+        'authors': [
+            'andela_bouwe',
+            'righi_mattia',
+        ],
+        'references': [
+            'acknow_project',
+        ],
```

Comment:
There is a link to "Provenance information" as part of the PR checklist; within that, the [Provenance items provided by the diagnostic script](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#provenance-items-provided-by-the-diagnostic-script) section provides keys that can be used in the provenance dictionary; for example, the `caption` is used for the plots in the HTML output from the recipe, so specifying this will replace the path below with the caption:

![image](https://github.com/ESMValGroup/ESMValTool/assets/918142/21f704e6-440f-47ae-8fbc-b0611c96f1d0)

Also, be sure that the `ancestor_files` contain only the files used for each plot! :)

Reply:
I've simplified the function get_provenance_record, added figure captions and fixed the ancestor files in b46c7924f08eb84d4397dee28bce147a5da6744e